File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Prison Cells After N Days.py
class Solution:
    def prisonAfterNDays(self, cells: List[int], N: int) -> List[int]:
        T = int(max(N-1, 0)/14)*14
        N -= T
        for i in range(N):
            cells = [0] + [cells[k-1]^cells[k+1]^1 for k in range(1, 7)] + [0]
        return cells

File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Prison Cells After N Days.rs
use std::cmp::max;
impl Solution {
    pub fn prison_after_n_days(cells: Vec<i32>, n: i32) -> Vec<i32> {
        let mut N = n;
        let mut T:i32 = max(N-1, 0)/14;
        N = N - T*14;
        let N = N as usize;
        let mut v = cells;
        for i in 0..N{
            let mut v2 = vec![0;8];
            v2[0] = 0;
            v2[7] = 0;
            for j in 1..7{
                v2[j] = (v[j-1] == v[j+1]) as i32;
            }
            v = v2;
        }
        return v;
    }
}